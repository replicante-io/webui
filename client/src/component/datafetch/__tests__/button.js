'use strict';
//@flow

import React from 'react';
import renderer from 'react-test-renderer';

import ReactTestUtils from 'react-dom/test-utils';
import configureStore from 'redux-mock-store';

import type { FetchStore } from '../store';
import Button from '../button';


const mockStore = configureStore([]);


describe('datafetch', () => {
  describe('Button', () => {
    const request = () => new Promise(() => {});
    let dispatch;
    let store: FetchStore;

    beforeEach(() => {
      dispatch = jest.fn();
      store = new Map();
    });

    test('renders correctly', () => {
      const tree = renderer.create(
        <Button id="test" dispatch={dispatch}
                store={store} request={request} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders error', () => {
      store.set('test', {id: 'test', active: true, error: 'error', success: false});
      const tree = renderer.create(
        <Button id="test" dispatch={dispatch}
                store={store} request={request} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders fetching', () => {
      store.set('test', {id: 'test', active: true, error: null, success: false});
      const tree = renderer.create(
        <Button id="test" dispatch={dispatch}
                store={store} request={request} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('renders success', () => {
      store.set('test', {id: 'test', active: true, error: null, success: true});
      const tree = renderer.create(
        <Button id="test" dispatch={dispatch}
                store={store} request={request} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('request on click', () => {
      const component = ReactTestUtils.renderIntoDocument(
        <Button id="test" dispatch={dispatch}
                store={store} request={request} />
      );
      // $FlowFixMe: flow mistakes this for a generic `React.Compoment`.
      ReactTestUtils.Simulate.click(component.button);
      expect(dispatch.mock.calls.length).toBe(1);
    });

  });
});
