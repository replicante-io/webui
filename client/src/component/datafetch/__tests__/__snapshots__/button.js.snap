// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datafetch Button renders correctly 1`] = `
<button
  className="btn btn-outline-secondary fetch-button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-redo-alt fa-w-16 "
    data-icon="redo-alt"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`datafetch Button renders error 1`] = `
<button
  className="btn btn-outline-danger fetch-button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-times fa-w-11 "
    data-icon="times"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 352 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`datafetch Button renders fetching 1`] = `
<button
  className="btn btn-outline-secondary fetch-button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-circle-notch fa-w-16 fa-spin "
    data-icon="circle-notch"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;

exports[`datafetch Button renders success 1`] = `
<button
  className="btn btn-outline-success fetch-button"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-check fa-w-16 "
    data-icon="check"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
